activerecord/lib/active_record/errors.rb

```diff
@@ -183,6 +183,10 @@ class PreparedStatementInvalid < ActiveRecordError
   class NoDatabaseError < StatementInvalid
   end
 
+  # Raised when a given table does not exist.
+  class NoTableError < ActiveRecordError
```

Comment:
Naming this `NoTableError` makes it sound equivalent to the existing `NoDatabaseError`, but they're actually a bit different: `NoDatabaseError` is raised when a database name was specified but doesn't exist, whereas this new error will only raised when no table name is specified.

It might be interesting to explore (in another PR) raising a more specific error class when the specified table doesn't exist, as we do for other known classes of database error:

https://github.com/rails/rails/blob/v5.2.3/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L636-L661
https://github.com/rails/rails/blob/v5.2.3/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L424-L441
https://github.com/rails/rails/blob/v5.2.3/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L496-L505

The natural name for that error class would be `NoTableError`, since it _would_ be equivalent to `NoDatabaseError`, but using the name here would preclude us from using it for that purpose later.

What do you think of naming this one `TableNotSpecified` instead, by analogy to [`AdapterNotSpecified`](https://github.com/rails/rails/blob/v5.2.3/activerecord/lib/active_record/errors.rb#L38)?

Reply:
Yes! I really like this, I couldn't quite put to words what I didn't like about this PR but I think thats it. `NoTableError` today is not a `ActiveRecord::StatementInvalid` exception like `NoDatabaseError` that comes form the DB but rather an app level error so I'm totally digging: `TableNotSpecified`. I'll update ❤️